src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -400,16 +404,411 @@ public void test_ecdh_1pu_a128kw_appendix() throws Exception {
+
+        OctetKeyPair bobKey = OctetKeyPair.parse("{\"kid\": \"bob-key-2\"," +
+                "         \"kty\": \"OKP\"," +
+                "         \"crv\": \"X25519\"," +
+                "         \"x\": \"BT7aR0ItXfeDAldeeOlXL_wXqp-j5FltT0vRSG16kRw\"," +
+                "         \"d\": \"1gDirl_r_Y3-qUa3WXHgEXrrEHngWThU3c9zj9A2uBg\"}");
+
+        OctetKeyPair charlieKey = OctetKeyPair.parse("{\"kid\": \"2021-05-06\"," +
+                "         \"kty\": \"OKP\"," +
+                "         \"crv\": \"X25519\"," +
+                "         \"x\": \"q-LsvU772uV_2sPJhfAIq-3vnKNVefNoIlvyvg1hrnE\"," +
+                "         \"d\": \"Jcv8gklhMjC0b-lsk5onBbppWAx5ncNtbM63Jr9xBQE\"}");
+
+        JWEObjectJSON jwe = JWEObjectJSON.parse("{" +
+                "     \"protected\":" +
+                "      \"eyJhbGciOiJFQ0RILTFQVStBMTI4S1ciLCJlbmMiOiJBMjU2Q0JDLUhTNTEyIiwiYXB1IjoiUVd4cFkyVSIsImFwdiI6IlFtOWlJR0Z1WkNCRGFHRnliR2xsIiwiZXBrIjp7Imt0eSI6Ik9LUCIsImNydiI6IlgyNTUxOSIsIngiOiJrOW9mX2NwQWFqeTBwb1c1Z2FpeFhHczluSGt3ZzFBRnFVQUZhMzlkeUJjIn19\"," +
+                "     \"unprotected\":" +
+                "      {\"jku\":\"https://alice.example.com/keys.jwks\"}," +
+                "     \"recipients\":[" +
+                "      {\"header\":" +
+                "        {\"kid\":\"bob-key-2\"}," +
+                "       \"encrypted_key\":" +
+                "        \"pOMVA9_PtoRe7xXW1139NzzN1UhiFoio8lGto9cf0t8PyU-sjNXH8-LIRLycq8CHJQbDwvQ" +
+                "        eU1cSl55cQ0hGezJu2N9IY0QN\"}," +
+                "      {\"header\":" +
+                "        {\"kid\":\"2021-05-06\"}," +
+                "       \"encrypted_key\":" +
+                "        \"56GVudgRLIMEElQ7DpXsijJVRSWUSDNdbWkdV3g012Nq6hcT_GkxwnxlPIWrTXCqRpVKQC8" +
+                "         fe4z3PQ2YH2afvjQ28aiCTWFE\"}]," +
+                "     \"iv\":" +
+                "      \"AAECAwQFBgcICQoLDA0ODw\"," +
+                "     \"ciphertext\":" +
+                "      \"Az2IWsISEMDJvyc5XRL-3-d-RgNBOGolCsxFFoUXFYw\"," +
+                "     \"tag\":" +
+                "      \"HLb4fTlm8spGmij3RyOs2gJ4DpHM4hhVRwdF_hGb3WQ\"" +
+                "    }");
+
+        try {
+            List<Pair<UnprotectedHeader, OctetKeyPair>> recipients = Arrays.asList(
+                    Pair.of(new UnprotectedHeader.Builder().keyID("bob-key-2").build(), bobKey),
+                    Pair.of(new UnprotectedHeader.Builder().keyID("2021-05-06").build(), charlieKey)
+            );
+
+            ECDH1PUX25519DecrypterMulti decrypterMulti = new ECDH1PUX25519DecrypterMulti(aliceKey.toPublicJWK(), recipients);
+            jwe.decrypt(decrypterMulti);
+            fail();
+        } catch (Exception e) {
+            assertEquals("Couldn't unwrap AES key: Integrity check failed", e.getMessage());
+        }
+
+        List<Pair<UnprotectedHeader, OctetKeyPair>> recipients = Collections.singletonList(
```

Comment:
A minor thing: I would make the second case (Bob's key is correct while Cahrlie's is not) a separate test method:
- it may be not immediately obvious w/o any comments why decryption failed in the first case and succeeded in the later case.
- It's better to follow best practices in writing tests, where every test method should test only one thing and have a common structure initialization -> execution -> assertion